### ISSUE TITLE
Reimplement python-rpm-macros for new rpmlua reflection api

### ISSUE THREAD

**Issue by bunzuhbu** [openSUSE/python-rpm-macros#189]
I think, if they implement that feature, described in discussion https://github.com/rpm-software-management/rpm/discussions/3484 , we could reimplement python-rpm-macros to make more stable and usable. Also, we can implement that feature with more other useful features, making a fork of rpm package manager, to outperform redhat on technical specifications.

**Comment by huakim** [comment]
Well, I think, that this is what we exactly need. @mcepl, is that better than rewriting specfile Python module to Lua?

**Comment by mcepl** [comment]
These macros must run unchanged on SLE-15. Does this work with rpm 4.14.1?

**Comment by bunzuhbu** [comment]
> These macros must run unchanged on SLE-15. Does this work with rpm 4.14.1?

Yes, it is

**Comment by bunzuhbu** [comment]
> These macros must run unchanged on SLE-15. Does this work with rpm 4.14.1?

what this?

**Comment by mcepl** [comment]
> > These macros must run unchanged on SLE-15. Does this work with rpm 4.14.1?
> 
> yes, python-rpm-macros works with rpm-4.14.1

I meant the suggested “new rpmlua reflection api”.

**Comment by bunzuhbu** [comment]
that api is not implemented yet, but SuSE can implement that to order to outperform redhat and take leadership on commercial linux

**Comment by bunzuhbu** [comment]
> > > These macros must run unchanged on SLE-15. Does this work with rpm 4.14.1?
> > 
> > 
> > yes, python-rpm-macros works with rpm-4.14.1
> 
> I meant the suggested “new rpmlua reflection api”.

that api is not implemented yet, but SuSE can implement that to order to outperform redhat and take leadership on commercial linux

**Comment by huakim** [comment]
renaming to suse-build is redundant, we can just create symbolic link to executable file rpm-build, named as suse-build, but improving that tool because of specific cases for SuSE (expecially when generating subpackages) is necessary. I looked at source code of python-rpm-macros, and that made me sad. Because in code just reimplemented rpm spec parser, and, moreover, when building from .spec file, containing %python_generate_subpackages, the macros expanded twice, which may lead to unexpected behavior.

**Comment by huakim** [comment]
the lua_generate_subpackages works in the same way as python_generate_subpackages, and, thesefore, we can not use them at the same time, which is certainly undesirable.

**Comment by huakim** [comment]
No, i do not think that forking rpm-build is out of the scope of this project, because, as i previously said, python_generate_subpackages does parse specfile, and, thesefore, macroses executed during rpm-build and during python_generate_subpackages, which lead unexpected behavior. I do not think that we need to implement just another spec file parser, just because we can not see internal structure and changes during rpm-build in lua, instead, it is better to fork rpm-build, add necessary features and then rewrite python-rpm-macros with using these features

**Comment by mcepl** [comment]
> No, i do not think that forking rpm-build is out of the scope of this project, …

And you would be wrong. `rpm-build` is part of `rpm` package and `rpm` project. Talk with them, and when you come to the agreement with those maintainers, you can return with a pull request here. Show me the code!

Meanwhile, …
